backend/src/main/java/quarano/actions/DiaryEventListener.java

```diff
@@ -23,14 +26,20 @@
 	private final @NonNull AnomaliesProperties config;
 	private final @NonNull ActionItemRepository items;
 	private final @NonNull DiaryManagement diaryManagement;
+	private final @NonNull TrackedCaseRepository cases;
+
 
 	@EventListener
 	void on(DiaryEntryAdded event) {
 
 		var entry = event.getEntry();
 
-		handleDiaryEntryForBodyTemprature(entry);
-		handleDiaryEntryForCharacteristicSymptoms(entry);
+		boolean isContactCase = cases.findByTrackedPerson(entry.getTrackedPersonId())
+				.map(TrackedCase::isContactCase)
+				.orElse(Boolean.TRUE);
+
+		handleDiaryEntryForBodyTemperature(entry, isContactCase);
+		handleDiaryEntryForCharacteristicSymptoms(entry, isContactCase);
```

Comment:
Technical details aside, I am wondering if what's done here maps whats requested by the original ticket. That says:

> First characteristic symptoms shall only be recorded for contact cases.

If I understand that correctly we could simply skip the `handleDiaryEntryForCharacteristicSymptoms(…)` in case that flag is true instead of handing it down. Correct?

Why do we need the flag for the `handleDiaryEntryForBodyTemperature(…)` call?

Reply:
My idea was that business logic in determineWhetherEntryMostRecent and findUnresolvedByDescriptionCode .. -> save

Are relevant for already existing entries (befor this story comes alive, entries exist for both types)

Reply:
> Can we extract this code to a method and use this in on(DiaryEntryAdded event) and in on(DiaryEntryUpdated event) to avoid duplicated code?

Good idea, applied.

> TrackedCase::isContactCase is the wrong method, because there are several types of contact. You must use isIndexCase.
Updated

I think we should invoke the methods handleDiaryEntryForBodyTemperature and handleDiaryEntryForCharacteristicSymptoms only for contact cases, which makes the parameter isContactCase unnecessary. 

findUnresolvedByDescriptionCode performs additional modifications on already existing action items. Those would exist before this version goes live.